arch/arm/src/armv7-a/cp15_clean_dcache_all.S

```diff
@@ -91,19 +91,30 @@
 
 cp15_clean_dcache_all:
 
+	push		{r4, r5}
+
 	mrc		CP15_CCSIDR(r1)			/* Read the Cache Size Identification Register */
+	mov		r5, r1
 
 	ldr		r3, =0x7fff			/* Isolate the NumSets field (bits 13-27) */
 	and		r0, r3, r1, lsr #13		/* r0=NumSets (number of sets - 1) */
 
 	ldr		r3, =0x3ff			/* Isolate the way field (bits 3-12) */
 	and		r1, r3, r1, lsr #3		/* r1=(number of ways - 1) */
 
+	add		r4, r1, #1
+	clz		r4, r4
+	add		r4, #1				/* r4=Set/way operation Way shift */
+
+	ldr		r3,=0x7				/* Isolate the LineSize field (bits 0-2) */
+	and		r5, r3				/* r4=(Log2LineSize - 2) in word */
```

Comment:
change to:
```
and r5, r3, r1
```
and remove line 97.

Reply:
apply the similar change to other place too.

Reply:
Done